src/TLuaInterpreter.cpp

```diff
@@ -6364,11 +6160,7 @@ int TLuaInterpreter::showUnzipProgress(lua_State* L)
 // Documentation: https://wiki.mudlet.org/w/Manual:Lua_Functions#playSoundFile
 int TLuaInterpreter::playSoundFile(lua_State* L)
 {
-    if (!lua_isstring(L, 1)) {
-        lua_pushstring(L, "playSoundFile: wrong argument type");
-        return lua_error(L);
-    }
-    QString sound = lua_tostring(L, 1);
+    QString sound = getVerifiedString(L, __func__, 1, "fileName");
 
```

Comment:
Why a down-thumb? Rather than giving even a "wrong argument type" error for a wrong 2nd argument it is currently being silently ignored if it is not a number - yet that is something that your fancy new functions could handle.